### PR TITLE
Add diversity data to UCAS export

### PR DESCRIPTION
**Pull request by aldavidson**
## Context

After discussion, we've agreed to supply the results of the equality_and_diversity survey to UCAS as part of "Dataset 1" - the CSV export we will send to UCAS every night so that they can detect duplicate applications/candidates in their system. This PR adds those fields.

## Changes proposed in this pull request

Add equality_and_diversity fields to the `applications_export_for_ucas` service
Add a `:with_equality_and_diversity_data` trait to the `:completed_application_form` factory
Add a utility method to the `EthnicBackgroundHelper` to generate all possible combinations of ethnic group & background (makes it much easier to pick a random-but-reasonable combination in test data)
Add the new trait to the arguments in `GenerateTestData`, for a representative 55% of candidates (that's roughly the percentage who have filled in the survey in production so far)
Add a FeatureFlag around the UI and controller action, so it won't be on production unless we explicitly enable it

## Guidance to review

1. Enable the 'download_dataset1_from_support_page' feature flag
2. From the console, run `GenerateTestData.new(10).generate`
3. Go to /support/performance
4. Scroll down to ''Applications export for UCAS"
5. Click 'Download Dataset 1 (CSV)'
6. Open the generated CSV file in the spreadsheet of your choice
7. You should see representative values in sex/ethnic background/ethnic group/disabilities etc for some (but not all - around 5 or 6) of the new applications 

## Link to Trello card

[UCAS data-matching - Dataset 1 changes requested by UCAS](https://trello.com/c/g6rVRm67/1318-ucas-data-matching-dataset-1-changes-requested-by-ucas)

## Things to check

- [x] This code doesn't rely on migrations in the same Pull Request
- [ ] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [ ] API release notes have been updated if necessary
- [ ] New environment variables have been [added to the Azure config](https://github.com/DFE-Digital/apply-for-postgraduate-teacher-training#azure-hosting-devops-pipeline)

### CODE REVIEW

File: app/helpers/ethnic_background_helper.rb [hunk 1]
```diff
@@ -48,4 +48,16 @@ def ethnic_background_title(group)
   def equality_and_diversity_caption
     '<span class="govuk-caption-xl">Equality and diversity</span>'.html_safe
   end
+
+  # return every possible combination of group + background
+  # used in the factory for generating sensible combinations in test data
+  def all_combinations
```
Comment (duncanjbrown) [inline app/helpers/ethnic_background_helper.rb:54]: `ETHNIC_GROUPS.product(ETHNIC_BACKGROUNDS)`? https://ruby-doc.org/core-2.6.6/Array.html#method-i-product

Comment (aldavidson) [reply app/helpers/ethnic_background_helper.rb:54]: I did initially try that, but couldn't get it to produce the output I needed (an array-of-arrays but no further nesting) without the code ending up at least as complex as doing it explicitly (`ETHNIC_BACKGROUNDS` is a Hash, so you have to dereference it to the particular group, and you can't do it as a straightforward map or you end up with 3-level-deep nesting....

That's not to say it's not possible, of course, just that I didn't want to waste too much time on it

Comment (duncanjbrown) [reply app/helpers/ethnic_background_helper.rb:54]: ah, I didn't realise it was a hash! trying to be clever, please ignore me

**Comment by aldavidson** [comment]
I've added a FeatureFlag around the UI & controller action, so it won't be on in Production unless we explicitly enable it

**Comment by tvararu** [comment]
Can confirm it works as expected on the Heroku review app using the provided steps. :+1

File: app/helpers/ethnic_background_helper.rb (hunk 1 above)
Comment (tvararu) [reply app/helpers/ethnic_background_helper.rb:54]: I got nerd sniped by this:

```ruby
ETHNIC_BACKGROUNDS
  .to_a
  .map {|(group, bgs)| ([group] * bgs.size).zip(bgs) }
  .flatten(1)
``` 

Works the same, but is vastly more arcane than the suggested code... yeah, stick to the nested loop!

**Comment by aldavidson** [comment]
@tvaruru happy for me to merge with the latest additions?

**Pull request merged**